### PR TITLE
[AE-51] Add documentation for BHY2Host

### PR DESCRIPTION
**Pull request by aliphys**
### Description

- Add in-code documentation for BHY2Host classes
    - Arduino_BHY2Host
    - BLEHandler
    - EslovHandler

### Pull request type

    [ ] Fix
    [x ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Breaking change

### CODE REVIEW

File: Arduino_BHY2Host/src/EslovHandler.h [hunk 1]
```diff
@@ -43,19 +57,80 @@ class EslovHandler {
...
+   */
   uint8_t requestAvailableData();
+  /**
+   * @brief Obtain long data sent over ESLOV to the host board
+   * 
+   * @return uint8_t Data read from I2C bus
+   */
   uint8_t requestAvailableLongData();
+  /**
+   * @brief Request data from the Nicla board
```
Comment (sebromero) [inline Arduino_BHY2Host/src/EslovHandler.h:113]: This is for a specific sensor, right? So do I need to pass the sensor ID or how does this work?

File: Arduino_BHY2Host/src/EslovHandler.h [hunk 2]
```diff
@@ -43,19 +57,80 @@ class EslovHandler {
...
+   * 
+   * @param sData 
+   * @return true   Request successful
+   * @return false  Request unsuccessful
```
Comment (sebromero) [inline Arduino_BHY2Host/src/EslovHandler.h:117]: Transmission error only or also configuration error?

File: Arduino_BHY2Host/src/EslovHandler.h [hunk 3]
```diff
@@ -43,19 +57,80 @@ class EslovHandler {
...
+   */
   bool requestSensorData(SensorDataPacket &sData);
+  /**
+   * @brief Request long data from the Nicla board
```
Comment (sebromero) [inline Arduino_BHY2Host/src/EslovHandler.h:121]: Same comment as for requestSensorData()

File: Arduino_BHY2Host/src/EslovHandler.h [hunk 4]
```diff
@@ -43,19 +57,80 @@ class EslovHandler {
...
+   * 
+   * @param sData 
+   * @return true  Request successful
+   * @return false Request unsuccessful
+   */
   bool requestSensorLongData(SensorLongDataPacket &sData);
 
 protected:
+  /**
+   * @brief Nicla Sense ME is mounted as a Shield
```
Comment (sebromero) [inline Arduino_BHY2Host/src/EslovHandler.h:131]: What does the function do? Does it set some pins, change some configuration, ...?

**Review by sebromero — CHANGES_REQUESTED** [review]

**Comment by aliphys** [comment]
Closing PR, changes are in https://github.com/arduino/nicla-sense-me-fw/pull/104

**Pull request closed without merging**